source/MRMesh/MRCone3.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "MRLine.h"
+
+namespace MR
+{
+
+// Base class for cone parameterization
+
+template <typename T>
+class Cone3
+{
+public:
+    Cone3()
+    {}
+
+    Cone3( const Line3<T>& inAxis, T inAngle, T inLength )
+        :
+        position( inAxis ),
+        angle( inAngle ),
+        length( inLength )
+    {}
+
+    inline MR::Vector3<T>& center( void )
+    {
+        return position.p;
+    }
+    inline MR::Vector3<T>& direction( void )
```

Comment:
Suggest adding comments: Does this point from the base to the apex? Is this guaranteed to be normalized?

Also need a second `const` overload for this and for `apex()` too.